### DISCUSSION TITLE
How do include pelephysics in my code

### DISCUSSION THREAD

**Discussion by bronise777** [AMReX-Combustion/PelePhysics#520] [Q&A]
I want to put pelephysics into submodules in my code, like pelec. What should my makefile do ?

**Comment by baperry2** [comment]
This depends a bit on what part of PelePhysics you are trying to use and for what purpose. There's not a simple universal example but you can look at what the other Pele codes do.

Assuming you are using the gnu make build system, the easiest example to follow might be the [PeleAnalysis Make.Analysis file](https://github.com/ThomasHowarth/PeleAnalysis/blob/e223c302868a438e31fc9284e52ca49f7348dd2e/Tools/GNUmake/Make.Analysis#L54) that @ThomasHowarth has cleaned up, which will work well if you need self-contained code from PelePhysics like what's in `Source/Utility`. If you need all of PelePhysics, including the reaction integraters, EOS, etc, then your best bet would probably be to study the [Make.PeleC file](https://github.com/AMReX-Combustion/PeleC/blob/development/Exec/Make.PeleC) and extract the PelePhysics pieces for use with your code.

**Reply by bronise777** [reply]
thank you~
